src/lib/kdf/kdf2/kdf2.cpp

```diff
@@ -11,7 +11,8 @@ namespace Botan {
 
 size_t KDF2::kdf(byte key[], size_t key_len,
                  const byte secret[], size_t secret_len,
-                 const byte salt[], size_t salt_len) const
+                 const byte salt[], size_t salt_len,
+                 const byte[], size_t) const
```

Comment:
I see what you mean. But for example IEEE 1363's KDF1 description just says "Key derivation parameters, which are an octet string P of length pLen octets." with no further requirements on how it is generated. Currently the Botan KDF interface accepts just a salt parameter, to set this value. So applications that want labels cram both label and random nonce into a single parameter (as also seen in TLS). Botan can reasonably define the KDF1/KDF2 key derivation parameters to be the concatenation of two arguments (``label || salt``), and the application can then set the KDF input via string (label arg) or binary vector (salt arg) or both.